### PR TITLE
Writes gob of service data to file

### PR DESCRIPTION
**Pull request by mieubrisse**
We're almost ready to Do The Thing!!!!!

### CODE REVIEW

File: initializer/test_suite_runner.go [hunk 1]
```diff
@@ -117,29 +124,33 @@ func (runner TestSuiteRunner) RunTests() (err error) {
...
 	if err != nil {
 		return stacktrace.Propagate(err, "Could not create tempfile to store network info for passing to test controller")
 	}
 
-	// TODO replace this with the actual JSON of the network
-	err = ioutil.WriteFile(tmpfile.Name(), []byte("JSON data would go here"), 0644)
+	logrus.Debugf("Temp filepath: %v", tmpfile.Name())
+
+	encoder := gob.NewEncoder(tmpfile)
+	println(fmt.Sprintf("%v", rawServiceNetwork))
```
Comment (galenmarchetti) [inline initializer/test_suite_runner.go:144]: can we replace this with a log statement?

File: initializer/test_suite_runner.go [hunk 2]
```diff
@@ -117,29 +124,33 @@ func (runner TestSuiteRunner) RunTests() (err error) {
...
+	err = encoder.Encode(rawServiceNetwork)
 	if err != nil {
-		return stacktrace.Propagate(err, "Could not write data to testing file")
+		return stacktrace.Propagate(err, "Could not write service network state to file")
 	}
 	// Apparently, per https://www.joeshaw.org/dont-defer-close-on-writable-files/ , file.Close() can return errors too,
```
Comment (galenmarchetti) [inline initializer/test_suite_runner.go:149]: so why defer the close? can we just call tmpfile.Close() and throw an error if closing fails?

**Review by galenmarchetti — CHANGES_REQUESTED** [review]

File: initializer/test_suite_runner.go (hunk 2 above)
Comment (mieubrisse) [reply initializer/test_suite_runner.go:149]: It's to make sure that a panic anywhere will still close the file (if a panic happens, a regular Close call won't get called). AFAICT, an error from Close is something you can't do anything about (I'm not even sure what the remediation steps would be - stopping the program seems wrong)

File: initializer/test_suite_runner.go (hunk 1 above)
Comment (mieubrisse) [reply initializer/test_suite_runner.go:144]: Fixed in latest PR

**Review by galenmarchetti — APPROVED** [review]

**Pull request merged**